sdk/tests/core/test_az_json.c

```diff
@@ -3105,17 +3105,6 @@ static void test_az_json_string_unescape(void** state)
     assert_true(az_span_is_content_equal(expected, destination_span));
   }
 
-  // insufficient space
-  {
-    az_span json = AZ_SPAN_FROM_STR(" { \"name\": \"some value string\" , \"code\" : 123456 } ");
-    char destination[5] = { 0 };
-    int final_size;
-
-    az_result result = az_json_string_unescape(json, destination, 4, &final_size);
```

Comment:
Strangely, this test for unescape doesn't have an escaped string.

I would have expected to see a test which had a string with escaped values where the destination was too short for the unescaped string.

Reply:
Agreed. That said, this existing test was only testing destination size being too small, independent of unescaping.